distributed/config.py

```diff
@@ -168,6 +172,74 @@ def initialize_logging(config):
             _initialize_logging_old_style(config)
 
 
+class RateLimitedLogger:
```

Comment:
I briefly entertained the idea of figuring out how to write a custom `logging.Handler` but quickly decided it was not worth the complication.